substrate-archive/src/actors/workers/blocks.rs

```
@@ -74,6 +74,29 @@ where
 		}
 	}
 
+	async fn fetch_genesis_block(&self) -> Result<Option<Block<B>>>
+	where
+		NumberFor<B>: From<u32>,
+	{
+		let backend = self.backend.clone();
+		if let Some(block) = backend.block(&BlockId::Number(0.into())) {
+			let cache = self.rt_cache.clone();
+			let spec = cache.get(block.block.hash())?.map(|ver| ver.spec_version).unwrap_or_default();
+			Ok(Some(Block { inner: block, spec }))
+		} else {
+			Ok(None)
```

Why not return an error here?